prov/shm/src/smr.h

```diff
@@ -275,9 +275,9 @@ size_t smr_copy_from_sar(struct smr_freestack *sar_pool, struct smr_resp *resp,
 			 struct smr_cmd *cmd, struct ofi_mr **mr,
 			 const struct iovec *iov, size_t count,
 			 size_t *bytes_done, int *next);
-int smr_select_proto(enum fi_hmem_iface, bool use_ipc, bool cma_avail,
-                     bool gdrcopy_avail, uint32_t op, uint64_t total_len,
-                     uint64_t op_flags);
+int smr_select_proto(enum fi_hmem_iface iface, void **desc, size_t iov_count,
+                     bool use_ipc, bool cma_avail, uint32_t op, uint64_t total_len,
```

Comment:
you shouldn't need to pass in iface or use_ipc anymore. this is all calculated locally now

Reply:
I think it would be better on the performance use parameters instead of re-calculating right?

Reply:
I'm not sure I'm following here. You're passing in use_ipc which now is no longer calculated in the calling function so there should be no use for it?

Reply:
scratch what I said... you are right.